pageserver/src/tenant/timeline/eviction_task.rs

```diff
@@ -271,19 +271,30 @@ impl Timeline {
                     stats.skipped_for_shutdown += 1;
                 }
                 Some(Ok(true)) => {
-                    debug!("evicted layer {l:?}");
+                    debug!("evicted layer {l}");
                     stats.evicted += 1;
                 }
                 Some(Ok(false)) => {
-                    debug!("layer is not evictable: {l:?}");
+                    debug!("layer is not evictable: {l}");
                     stats.not_evictable += 1;
                 }
                 Some(Err(e)) => {
-                    // This variant is the case where an unexpected error happened during eviction.
-                    // Expected errors that result in non-eviction are `Some(Ok(false))`.
-                    // So, dump Debug here to gather as much info as possible in this rare case.
-                    warn!("failed to evict layer {l:?}: {e:?}");
-                    stats.errors += 1;
+                    let already_deleted = e
+                        .root_cause()
+                        .downcast_ref::<std::io::Error>()
+                        .filter(|io| io.kind() == std::io::ErrorKind::NotFound)
+                        .is_some();
+
+                    if already_deleted {
+                        debug!("evicted already deleted layer: {l}");
+                        stats.evicted += 1;
+                    } else {
```

Comment:
Well, this is silly, it would have to be handled in all call sites. Better to involve it in the `layer_batch_evict_impl`.

Reply:
Complete rewrite after this.